augur/validate.py

```diff
@@ -53,17 +53,6 @@ def load_json_schema(path, refs=None):
     else:
         schema_validator = Validator(schema)
 
-    # By default $ref URLs which we don't define in a schema_store are fetched
-    # by jsonschema. This often indicates a typo (the $ref doesn't match the key
-    # of the schema_store) or we forgot to add a local mapping for a new $ref.
-    # Either way, Augur should not be accessing the network. 
-    def resolve_remote(url):
```

Comment:
> I'm not sure what could trigger this exception, but it doesn't seem to be used for its intended purpose.

We need this code to avoid Augur making requests over the network which I don't want. I'm not sure whether overwriting this method in v4 is still possible or the right approach.

If you (e.g.) comment out any of these lines you'll get this error raised when you validate an Auspice JSON:

https://github.com/nextstrain/augur/blob/e90383bf99e4d6758a40c9140998be63f186d6ef/augur/validate.py#L196-L200


You also get the error raised if you, e.g., change the $ref URI here:

https://github.com/nextstrain/augur/blob/e90383bf99e4d6758a40c9140998be63f186d6ef/augur/data/schema-export-v2.json#L40-L42


```console
jsonschema.exceptions.RefResolutionError: The schema used for validation attempted to fetch the remote URL 'https://nextstrain.org/schemas/augur/annotation_BAD'. Augur should resolve schema references to local files, please check the schema used and update the appropriate schema_store as needed.
```

Reply:
Thanks, I hadn't tested those two scenarios. I thought there wasn't a way to override the function, but looks like there is. Note that the default [no longer access the network](https://github.com/python-jsonschema/referencing/blob/2047eca80aa46c4e6dd5751cf669cc74233b5ecb/referencing/_core.py#L324) so having a custom function isn't necessary anymore, but I've kept it to show a custom error (though it's slightly hacky).

https://github.com/nextstrain/augur/blob/aeb807d588ab93c1de7128d56d1dd60d3ef97e6b/augur/validate.py#L54-L66

Reply:
The main aim of overriding that method was to prevent access over the wire. If that's now the default then consider this error optional - I think it'll make debugging easier, but I haven't played with the v4 API.